powsybl/powsybl-case-server
# disable auto case indexation

**Please check if the PR fulfills these requirements**



**What kind of change does this PR introduce?**
this PR disable auto indexation of the cases, and intreduce a new param "indexed" to indicate wich cases should be indexed by` /v1/supervision/case/reindex`
This also PR create a supervision service to delete elk documents and reindex cases.


**What is the current behavior?**
when a case is created or duplication , this service index it automatically 



**What is the new behavior (if this is a feature change)?**
cases are indexed on demand


**Does this PR introduce a breaking change or deprecate an API?**


**If yes, please check if the following requirements are fulfilled**

- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration steps are described in the following section

**What changes might users need to make in their application due to this PR? (migration steps)**
The endpoint for reindexing all cases has been renamed from `/v1/cases/index-all` to `/v1/supervision/cases/reindex`. 
Any client applications or scripts that previously called the `/v1/cases/index-all` endpoint will need to be updated to use the new `/v1/supervision/cases/reindex` endpoint.

The database schema has also been changed to add a new column "indexed" , the values of this collumn will be set to false by default.


**Other information**:


To test this PR : 
- delete elastic search index "cases
- restart the service
- use the endpoint `/v1/supervision/cases/reindex` to recreate the index ( you should have no cases indexed , all of them are marked indexed=false by default)
- choose some cases and update "indexed" value using `PUT /cases/{caseUuid}/indexation` 
- re-do a delete index - re-create index , now the choosen cases should appear in elastic search index
